src/Microsoft.VisualStudio.Threading.Tests/AsyncReaderWriterLockTests.cs

```diff
@@ -5013,18 +5013,17 @@ protected override Exception OnCriticalFailure(Exception ex)
                 throw new CriticalErrorException(ex);
             }
 
-            internal struct InternalLockHandle
+            internal readonly struct InternalLockHandle
```

Comment:
Wow. You're even optimizing _test_ structs? Did you have an analyzer that did all this for you?

Reply:
Looks like ErrorProne.NET https://blogs.msdn.microsoft.com/seteplia/2018/05/03/avoiding-struct-and-readonly-reference-performance-pitfalls-with-errorprone-net/